readthedocs/doc_builder/python_environments.py

```diff
@@ -193,12 +195,18 @@ def is_obsolete(self):
         # (e.g. ``2`` or ``3``) we won't know exactly which exact version was
         # used to create the venv but we can still compare it against the new
         # one coming from the project version config.
+        print(envvars_hash, self._get_envvars_hash())
         return any([
             env_python_version != self.config.python_full_version,
             env_build_image != build_image,
             env_build_hash != image_hash,
+            envvars_hash != self._get_envvars_hash(),
         ])
 
+    def _get_envvars_hash(self):
+        env_vars = self.version.project.environmentvariable_set.values_list('name', 'value')
```

Comment:
can you add a docstring for this method, please?

Reply:
Thanks.
I have pushed the changes.